pallets/collator-selection/src/weights.rs

```diff
@@ -39,91 +39,143 @@ pub trait WeightInfo {
 	}
 	fn set_candidacy_bond() -> Weight {
-		(16_840_000 as Weight).saturating_add(T::DbWeight::get().writes(1 as Weight))
+		Weight::from_ref_time(16_840_000 as RefTimeWeight)
+			.saturating_add(T::DbWeight::get().writes(1 as RefTimeWeight))
 	}
 	fn register_as_candidate(c: u32) -> Weight {
-		(71_196_000 as Weight)
+		Weight::from_ref_time(71_196_000 as RefTimeWeight)
 			// Standard Error: 0
-			.saturating_add((198_000 as Weight).saturating_mul(c as Weight))
-			.saturating_add(T::DbWeight::get().reads(4 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(
+				Weight::from_ref_time(198_000 as RefTimeWeight)
+					.scalar_saturating_mul(c as RefTimeWeight),
+			)
+			.saturating_add(T::DbWeight::get().reads(4 as RefTimeWeight))
+			.saturating_add(T::DbWeight::get().writes(2 as RefTimeWeight))
 	}
 	fn leave_intent(c: u32) -> Weight {
-		(55_336_000 as Weight)
+		Weight::from_ref_time(55_336_000 as RefTimeWeight)
 			// Standard Error: 0
-			.saturating_add((151_000 as Weight).saturating_mul(c as Weight))
-			.saturating_add(T::DbWeight::get().reads(1 as Weight))
-			.saturating_add(T::DbWeight::get().writes(2 as Weight))
+			.saturating_add(
+				Weight::from_ref_time(151_000 as RefTimeWeight)
+					.scalar_saturating_mul(c as RefTimeWeight),
+			)
+			.saturating_add(T::DbWeight::get().reads(1 as RefTimeWeight))
+			.saturating_add(T::DbWeight::get().writes(2 as RefTimeWeight))
 	}
 	fn note_author() -> Weight {
-		(71_461_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(3 as Weight))
-			.saturating_add(T::DbWeight::get().writes(4 as Weight))
+		Weight::from_ref_time(71_461_000 as RefTimeWeight)
+			.saturating_add(T::DbWeight::get().reads(3 as RefTimeWeight))
+			.saturating_add(T::DbWeight::get().writes(4 as RefTimeWeight))
 	}
 	fn new_session(r: u32, c: u32) -> Weight {
-		(0 as Weight)
+		Weight::from_ref_time(0 as RefTimeWeight)
 			// Standard Error: 1_010_000
-			.saturating_add((109_961_000 as Weight).saturating_mul(r as Weight))
+			.saturating_add(
+				Weight::from_ref_time(109_961_000 as RefTimeWeight)
+					.scalar_saturating_mul(r as RefTimeWeight),
+			)
 			// Standard Error: 1_010_000
-			.saturating_add((151_952_000 as Weight).saturating_mul(c as Weight))
-			.saturating_add(T::DbWeight::get().reads((1 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().reads((2 as Weight).saturating_mul(c as Weight)))
-			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(r as Weight)))
-			.saturating_add(T::DbWeight::get().writes((2 as Weight).saturating_mul(c as Weight)))
+			.saturating_add(
+				Weight::from_ref_time(151_952_000 as RefTimeWeight)
+					.scalar_saturating_mul(c as RefTimeWeight),
+			)
+			.saturating_add(
+				T::DbWeight::get().reads((1 as RefTimeWeight).saturating_mul(r as RefTimeWeight)),
+			)
+			.saturating_add(
+				T::DbWeight::get().reads((2 as RefTimeWeight).saturating_mul(c as RefTimeWeight)),
```

Comment:
We could think about a DLS for the weight annotations in the long run.  
This is starting to look really unreadable - at least for humans.

Reply:
i think making the weight file structure into a rust macro is not such a bad idea